src/Player/PlayerHackKeeper.h

```diff
@@ -1,20 +1,77 @@
 #pragma once
 
-#include <al/Library/LiveActor/LiveActor.h>
+#include <math/seadMatrix.h>
+#include <container/seadPtrArray.h>
+
+namespace al {
+    class LiveActor;
+    class HitSensor;
+    class CollisionPartsFilterBase;
+} // namespace al
 
 class PlayerRecoverySafetyPoint;
 class HackCap;
 class PlayerInput;
 class PlayerDamageKeeper;
 class IPlayerModelChanger;
 class IUsePlayerHeightCheck;
+class HackObjInfo;
+class CapTargetInfo;
+class PlayerHackStartTexKeeper;
 
 class PlayerHackKeeper {
+private:
+    al::LiveActor* mParent;
+    HackCap* mHackCap;
+    PlayerRecoverySafetyPoint* mRecoverySafePoint;
+    void* field_18;
+    void* field_20;
+    PlayerInput* mInput;
+    sead::Matrix34f* field_30;
+    PlayerDamageKeeper* mDamageKeeper;
+    IPlayerModelChanger* mModelChanger;
+    IUsePlayerHeightCheck* mHeightCheck;
+    al::HitSensor* mParentBodySensor;
+    bool mIsPuppetable;
+    bool mIsCancellingHack;
+    bool mIsHackDemoStarted;
+    bool mIsPuppetable2;
+    bool mIsStartedHacking;
+    bool mIsHack;
+    bool mIsTookDamage;
+    al::CollisionPartsFilterBase* mCollisionFilter;
+    al::LiveActor* mHackActor;
+    al::HitSensor* mHackHitSensor;
+    HackObjInfo* mHackObjectInfo;
+    al::HitSensor* mStageStartActorSensor;
+    al::LiveActor* mStageStartActor;
+    CapTargetInfo* mStageStartCapTargetInfo;
+    PlayerHackStartTexKeeper* mHackStartTexKeeper;
+    al::LiveActor* mHackModel;
+    sead::PtrArray<void> mHackModelSlices;
+    sead::Matrix34f* field_b8;
+    sead::Vector3i mLastPosition; // unsure
```

Comment:
Position as `float` seems weird. Can you point out any function that would show this type?

Reply:
its just a set of 3 `int`s that appear to be used together, so i thought it might be a vector... but i don't know for sure.

Reply:
If it is the set of three `int`s in `createHackModel`, I see nothing related to a position... not sure if my struct is correctly copied though.

Reply:
its probably just 3 ints that are loosely related. i'll just change them to 3 ints with no name.